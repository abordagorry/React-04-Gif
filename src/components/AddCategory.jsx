import { useState } from "react"
import { PropTypes } from "prop-types";

export const AddCategory = ( { onNewCategory } ) => {
    const [inputValue,  setInputValue]  = useState('')

    const onInputChange =( { target } ) =>{
        //setInputValue( 'Hola Mundo' )
        setInputValue( target.value )
        // console.log( event.target.value )
    }

    const onSubmit=(event)=>{
        event.preventDefault()
        //setCategories( [ inputValue ] )
        if( inputValue.trim().length <= 0 ) return
        // setCategories( categories => [inputValue, ...categories ] )
        setInputValue( '' )
        onNewCategory( inputValue.trim() )
    }

    return (
    <form onSubmit={ onSubmit } aria-label="form">
        <input
            type="text"
            placeholder="Buscar Gif"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes ={
    onNewCategory: PropTypes.func.isRequired,
}

