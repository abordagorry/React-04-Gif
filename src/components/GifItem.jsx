

export const GifItem = ( { images } ) => {
  return (
    <div className="card">
        <img src={ images.url } alt={ images.title }></img>
        <p>{ images.title }</p>
    </div>
  )
}
