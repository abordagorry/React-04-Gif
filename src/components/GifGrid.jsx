// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ( { category } ) => {
    // const [counter, setCounter] = useState(10)
    // const [images, setImages] = useState([])

    // const getImages = async() =>{
    //     const newImages = await getGifs( category );
    //     setImages( newImages );

    // }
    // useEffect( ()=>{
    // //    getGifs( category );
    //     getImages()
    // }, [] 
    // )
    //CAMBIAMOS EL CODIGO ANTERIOR POR UN HOOK

    const { images, isLoading } = useFetchGifs( category );

    return (
    <>
        <h3>{ category }</h3>
        {
            isLoading && ( <h2 >Cargando...</h2> )
        }
        
        {/* <button onClick={ () => setCounter( counter + 1 )} >+ 1 { counter }</button> */}
        
        <div className='card-grid'>
            {
            images.map( ( images ) => (
                // <li key={ id }>{ title  }</li> 
                <GifItem 
                    key={ images.id } 
                    images={ images }
                    //esparcir propiedades
                    // { ...images }
                /> 
            ))
            }
        </div>
    </>
  )
}

