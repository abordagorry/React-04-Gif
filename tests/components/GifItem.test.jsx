import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas <GiftItem />', () => { 
    const title = 'Saitama'
    const url   = 'https://gmail.com/1.jpg'

    test('Debe hacer match con snapshot', () => { 
        const { container } = render( <GifItem title={ title } url={ url } /> )
        expect( container ).toMatchSnapshot();

     }) 

     test('Debe mostar la imagen con el url y el ALT indicado', () => { 
        render( <GifItem title={ title } url={ url } /> )
        //screen.debug() //MUSTRA EL SNAP CRADO
// console.log( screen.getByRole( 'img' ))
// console.log( screen.getByRole( 'img' ).pendingProps.src)

// console.log( screen.getByRole( 'img' ).alt)

        //Una forma de comparar 
        expect( screen.getByRole( 'img' ).src ).toBe( url )
        expect( screen.getByRole( 'img' ).alt ).toBe( title )
 
        //Forma con desetructuracion
        const { src, alt } = screen.getByRole( 'img' )
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
      })

      test('Debe mostrar el titulo en el componente', () => { 
        render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title ) ).toBeTruthy

       })

 })