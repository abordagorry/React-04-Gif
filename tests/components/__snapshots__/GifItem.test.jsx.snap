// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GiftItem /> Debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://gmail.com/1.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
